FurkannOzbek/application-hub
# [FIX] Improve Responsiveness of Application Board with Mobile Optimization

# [FIX] Improve Responsiveness of Application Board with Mobile Optimization

---

## Description

This PR enhances the application tracking board with responsive design optimization, providing a better user experience across all devices. The main focus is on improving mobile usability by implementing an accordion-style view for smaller screens while maintaining the kanban board layout for desktop/tablet views.

- Problem: The original kanban board layout was not optimal for mobile devices
- Impact: Improved user experience on mobile devices with better content organization and accessibility

---

## Changes Made

Frontend Architecture
- Split ApplicationsBoard into separate components for better maintainability:
   - MobileApplicationsBoard: New accordion view for mobile
   - DesktopApplicationsBoard: Existing kanban board layout
   - Moved board-related components to dedicated /board directory

UI/UX Improvements
- Mobile View:
   - Implemented accordion-style navigation for status columns
   - Change colours of statuses
   - Optimized spacing and typography for mobile screens
- Desktop View:
   - Refined grid layout and spacing
   - Improved column responsiveness
   - Adjusted SearchField opacity and placeholder styles 
   - Change card styling
   - Added new theme colours for dashboard backgrounds

---

## Screenshots (if applicable)

1.	Mobile view with accordions
<img width="574" alt="Screenshot 2024-11-16 at 10 24 49" src="https://github.com/user-attachments/assets/819ba850-ad6f-458d-9aa3-ed378c193ece">

<img width="574" alt="Screenshot 2024-11-16 at 10 25 01" src="https://github.com/user-attachments/assets/c148dd6c-a556-4e27-acdf-123ec5fa17dd">

<img width="551" alt="Screenshot 2024-11-16 at 10 24 05" src="https://github.com/user-attachments/assets/d9764f84-0733-44a1-92b3-e24056edd11e">

<img width="578" alt="Screenshot 2024-11-16 at 10 24 34" src="https://github.com/user-attachments/assets/38dddd19-6d1c-43e1-a33c-6a06ed091c88">


2.	Desktop view with columns

<img width="912" alt="Screenshot 2024-11-16 at 10 23 36" src="https://github.com/user-attachments/assets/c14fc23c-7b80-49dc-a198-a9cd13e40e52">

<img width="914" alt="Screenshot 2024-11-16 at 10 23 48" src="https://github.com/user-attachments/assets/2959071c-001f-4371-92e8-6aae9e31cf56">

---

## Testing Steps

1. Mobile View Testing:
- Open application on mobile device or use device emulator
- Verify accordion functionality for each status
- Check application cards display correctly
- Test counter chips accuracy
2. Desktop View Testing:
- Verify kanban board layout on larger screens
- Test column responsiveness at different breakpoints
